pandas/core/frame.py

```diff
@@ -2586,7 +2591,29 @@ def transpose(self, *args, **kwargs):
         dtype: object
         """
         nv.validate_transpose(args, dict())
-        return super().transpose(1, 0, **kwargs)
+        # construct the args
+
+        dtypes = list(self.dtypes)
+        if self._is_homogeneous_type and dtypes and is_extension_array_dtype(dtypes[0]):
+            # We have EAs with the same dtype. We can preserve that dtype in transpose.
+            dtype = dtypes[0]
+            arr_type = dtype.construct_array_type()
+            values = self.values
+
+            new_values = [arr_type._from_sequence(row, dtype=dtype) for row in values]
+            result = self._constructor(
+                dict(zip(self.index, new_values)), index=self.columns
+            )
+
+        else:
+            new_values = self.values.T
+            if copy:
+                new_values = new_values.copy()
```

Comment:
if non-homogeneous, then `new_values` above is already a copy, can avoid re-copying here

Reply:
Are we 100% sure about that? Or are there types distinct dtypes that `.values` can combine without copy?

Reply:
if non-homogeneous then we have multiple blocks, so multiple ndarrays that are going through `np.c_` or something like it, right?  AFAIK that has to allocate new data for the output.  are there corner cases were missing @shoyer?